src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs

```diff
@@ -198,7 +198,7 @@ internal void NormalizeElementKernelNames(KernelInfoCollection kernelInfos)
 
     public string? GetDefaultKernelName()
     {
-        if (TryGetKernelInfoFromMetadata(Metadata, out var kernelInfo))
+        if (TryGetKernelInfosFromMetadata(Metadata, out var kernelInfo))
         {
             return kernelInfo.DefaultKernelName;
         }
```

Comment:
Why is the public API missing the remaining heuristics (like checking for` kernelspec` metadata)